### PR TITLE
Mobile App Upsell on Landing Page

### PR DESCRIPTION
**Pull request by jakzaizzat**
### Summary of Changes

Provide a brief description of the changes made in this PR. Explain the problem you are fixing or the feature you are adding.

### Demo or Before/After Pics

- If this is a new feature, include screenshots or recordings of the feature in action.
- If this PR makes visual changes, include before and after screenshots.

| Before                                     | After                                      |
| ------------------------------------------ | ------------------------------------------ |
|  |  |

### Edge Cases

List any common edge cases that you have considered and tested.

### Testing Steps

Provide steps on how the reviewer can test the changes.

### Checklist

Please make sure to review and check all of the following:

- [ ] I've tested the changes and all tests pass.
- [ ] (if web) I've tested the changes on various desktop screen sizes to ensure responsiveness.
- [ ] (if mobile) I've tested the changes on both light and dark modes.

### CODE REVIEW

File: apps/web/src/contexts/bottomsheet/BottomSheetContext.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  const context = useContext(BottomSheetActionsContext);
+  if (!context) {
+    throw new Error('Attempted to use BottomSheetModalActionsContext without a provider!');
+  }
+  return context;
+};
+
+type BottomSheetModalProviderProps = {
+  children: React.ReactNode;
+};
```
Comment (Robinnnnn) [inline apps/web/src/contexts/bottomsheet/BottomSheetContext.tsx:36]: hypernit: for situations like this, we can actually just use `PropsWithChildren` and import directly from react: `import { PropsWithChildren } from 'react';`

probably a few places to refactor in our repo, no need to handle in this PR :)

Comment (Robinnnnn) [reply apps/web/src/contexts/bottomsheet/BottomSheetContext.tsx:36]: oh maybe this will mess with the `BottomSheet` type you defined below, ignore if so

**Review by Robinnnnn — APPROVED** [review]

File: apps/web/src/contexts/bottomsheet/BottomSheetContext.tsx (hunk 1 above)
Comment (jakzaizzat) [reply apps/web/src/contexts/bottomsheet/BottomSheetContext.tsx:36]: Seems like it's messed up with the types 🤦🏻‍♂️
